tests/integration/customizations/s3/test_plugin.py

```diff
@@ -1055,6 +1055,33 @@ def test_bad_symlink(self):
                       p.stderr)
 
 
+class TestErrorAndWarning(BaseS3IntegrationTest):
```

Comment:
What is your thoughts on moving this to a functional test? It would be better because we run the functional tests more frequently and we do not necessarily need to send the request over the wire to test for this.

Reply:
I originally looked into implementing this as a functional test but opted for an integration test because I wasn't sure if I could guarantee the order that the requests are made. All examples of the functional tests for these currently only have a single line of api request logic and not two. Basically, will the S3 transfer manager guarantee the order of the `put_object` calls?

Reply:
So you could set the `max_concurrent_requests` rate to `1` to avoid that and it is fine that we would be overriding the config file because we would not be making an API requests. From there you could recreate a scenario that would trigger this. There are a probably a handful that you can do but one would be to do a recursive upload and make one of the files a special file type so it gets skipped and the other be normal and have the `PutObject` fail. This way you do not have to override the `max_concurrent_requests` either. The one downside is that we would have to skip for Windows with this approach specifically for the special file. But if you can think of something that does not skip Windows that would be ideal, maybe look around in the tests for cases where we skip?

Reply:
Actually instead of a special file, you could use the `set_invalid_utime` from before and that should allows us to avoid skipping on Windows.

Reply:
Looks like this is just pending this feedback getting incorporated right?  Where did we land on this?

Reply:
Yep, gonna update this to become a functional test.